webapp/packages/plugin-sql-editor-new/src/SQLEditor/SQLCodeEditorPanel/useSQLCodeEditorPanel.ts

```diff
@@ -44,13 +50,29 @@ export function useSQLCodeEditorPanel(data: ISQLEditorData, editor: IEditor) {
   );
 
   const updateHighlight = useCallback(
-    throttle(() => state.highlightActiveQuery(), 1000),
+    debounce(() => state.highlightActiveQuery(), 300),
```

Comment:
It is more performance and UX-friendly. Now with 120k rows in SQL editor I have almost no lags on my M1 PRO